docs/001_develop/03_client-capabilities/023_custom-components/03_state-management

```diff
@@ -0,0 +1,342 @@
+
+```shell
+import { attr, customElement, GenesisElement, html } from '@genesislcap/web-core';
+
+const template = html<MyButton>`
+    <button ?disabled="${x => x.isDisabled}" @click="${x => x.handleClick}">
+        ${x => x.label}
+    </button>
+`;
+
+@customElement({
+  name: 'my-button',
+  template,
+  styles,
+})
+
+export class MyButton extends FASTElement {
+    @attr label = 'Click Me';
+    @attr({ mode: 'boolean' }) isDisabled = false;
+
+    handleClick() {
+        if (!this.isDisabled) {
+            console.log('Button clicked');
+        }
+    }
+}
+```
+
+#### Usage in HTML
+
+```shell
+<my-button label="Submit" disabled></my-button>
+```
+
+- The ?disabled="${x => x.isDisabled}" syntax lets the templating engine handle boolean attributes automatically.
+  - If isDisabled is true, the disabled attribute is added to the button (`<button disabled>`).
+  - If isDisabled is false, the disabled attribute is removed from the button.
+- The `@attr({ mode: 'boolean' })` decorator ensures that changes to isDisabled in JavaScript reflect in the DOM as a disabled attribute, and vice versa.
+
+## Properties (@attr)
+
+It's important to distinguish between attributes and properties. Attributes are part of the HTML markup. They provide initial configuration and are typically static values of type string. Properties on the other hand are JavaScript-based, representing the internal state and can be any data type (boolean, number, object, etc)
+
+Attributes can be linked to properties via reflection, so updates to one (e.g., setting a property) can update the other (e.g., the attribute in HTML) and vice versa.
+
+We can manage reflection using the `@attr` decorator, making attributes and properties automatically sync to reflect the latest state
+
+> Properties and attributes are effectively two sides of the same coin. The @attr decorator creates a two-way binding between a JavaScript property on the component and a DOM attribute. This means that changes to the attribute in the HTML will reflect on the property in JavaScript, and changes to the property in JavaScript will update the attribute in the DOM.
+
+While it's important to distinguish between an attribute and a property, it's just as important to know the following about properties:
+
+- they can be defined internally in the component class without linking them to HTML attributes. These internal properties are not exposed in the DOM as attributes.
+
+> clickCount is an internal property and it’s only accessible within the component and is not reflected in the HTML markup. Only label is accessible as an HTML attribute.
+
+```shell
+import { attr, customElement, GenesisElement, html } from '@genesislcap/web-core';
+
+export class MyButton extends GenesisElemen {
+    @attr label = "Click Me"; // Linked attribute for label
+    clickCount = 0; // Internal property, not linked to an attribute
+
+    handleClick() {
+        this.clickCount++; // Update internal state
```

Comment:
Very much a nitpick, but again our linting rules prohibit the `++` and `--` operators.
```suggestion
        this.clickCount += 1; // Update internal state
```

Reply:
The changes to this example need to additionally be applied to the following examples on this page too.